pkg/kubelet/kubelet.go

```diff
@@ -535,6 +535,13 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		keepTerminatedPodVolumes:                keepTerminatedPodVolumes,
 	}
 
+	if klet.cloud != nil {
+		klet.cloudproviderRequestParallelism = make(chan int, 1)
+		klet.cloudproviderRequestSync = make(chan int)
+		// TODO(jchaloup): Make it configurable via --cloud-provider-request-timeout
+		klet.cloudproviderRequestTimeout = 3 * time.Minute
```

Comment:
how did you settle on 3m?

i feel like this should be less than the heartbeat status interval

Reply:
I agree, either 10s (status frequency) or 30s (10s less than the 40s timeout before the node is marked NotReady. I favor 10s.

There is also the fundamental issue of do we need to do this call on every status update.  Seems like this information (hostame, internal ip, external ip) is pretty static.

Reply:
> how did you settle on 3m?

for testing purposes. I plan to make the timeout configurable. Or does it make more sense to hardcode it?